### DISCUSSION TITLE
StackPanel with ItemsControl vs Combo Box for selection of specific object

### DISCUSSION THREAD

**Discussion by reelthymeos** [AvaloniaUI/Avalonia#17732] [Q&A] [answered]
Hello,

I have a scenario where my Main View model is going to have a list of objects that I call **Instrument** type. I want to display that list to the user and allow them to select a specific object from that list. The selected object will then be used elsewhere in code. 

I thought that an elegant way to do this is to have a stack panel with Items control inside. Each item in the stack panel will display the Instrument name and a checkbox next to the name so the user can select that Instrument if they want. Below is the code for the view of that:

```
<StackPanel Grid.Row="1" Margin="5">
	<ItemsControl ItemsSource="{Binding InstrumentList}">
		<ItemsControl.ItemTemplate>
			<DataTemplate>
				     <StackPanel Orientation="Horizontal" Grid.Column="0">
						<CheckBox HorizontalAlignment="Center" Margin="2" IsChecked="{Binding Selected}"/>
						<TextBlock VerticalAlignment="Center" Text="{Binding Name}"/>
				   </StackPanel>
			</DataTemplate>
	       </ItemsControl.ItemTemplate>
	</ItemsControl>
</StackPanel>

```

The type **Instrument** has a boolean property called "Selected." The idea is that I want a way to indicate in the ViewModel whether a given object in the list is the selected one. The problem is, I only want to allow one instrument in this list to be selectable at a time. It should be exclusive. I know how to do that in a collection of radio buttons for example, where you can lump them into a group and therefore say that only one of them can be selected at a time.  But here I am using an Items Control with Data template, so the list is being displayed dynamically. Ultimately my main view needs to know which Instrument instance is the selected one. 

A less elegant solution I thought would be to simply use a combo box,  and utilize the SelectedItem property to let the ViewModel know which object is the one to use. 

Thoughts?

**Comment by timunie** [comment, marked as answer]
What's wrong with a ListBox?

**Reply by reelthymeos** [reply]
That could work, that would be pretty much equivalent to the Combobox route I was considering.
